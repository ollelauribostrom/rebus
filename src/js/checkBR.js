export function checkResetPTBR(param) {
  let resetIdiom = param;
  if (localStorage.getItem('flagBR') === 'true') {
    resetIdiom = 'reset-ptbr';
  } else {
    resetIdiom = 'reset-english';
  }
  return resetIdiom;
}

export function checkIsPTBR() {
  if (localStorage.getItem('flagBR') === 'true') {
    document.querySelector('.root').classList.add('rootbr');
    document.getElementById('button-ptbr').classList.add('button-spam');
  } else {
    document.getElementById('button-en').classList.add('button-spam');
  }
}
