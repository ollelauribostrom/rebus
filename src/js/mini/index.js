export { createComponent } from './component';
export { render } from './render';
export { createStore } from './store';
