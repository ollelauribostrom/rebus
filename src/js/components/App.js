import { createComponent } from '../mini';

export function App(...children) {
  return createComponent({
    children,
    render() {
      return `
        <div class="app">
          <children>
        </div>
      `;
    }
  });
}
