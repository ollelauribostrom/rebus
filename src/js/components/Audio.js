import { createComponent } from '../mini';
export function AudioPlayer(props) {
  return createComponent({
    props,
    render({ src =''}) {
      return `
      <div class = "audio">
      <button>
      <audio  id = "audio" src= ${src} controls autoPlay={false}></audio>
        Music
        </button>
      </div>
     
      `;
    }
  });
}
