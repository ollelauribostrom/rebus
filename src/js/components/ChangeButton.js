import { createComponent } from '../mini';

export function ChangeButton(props) {
  return createComponent({
    props,
    render({ className = '' }) {
      return `
        <button class="change-button ${className}">
          <svg width="36px" height="67px" viewBox="0 0 36 67" version="1.1" xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink">
            <defs></defs>
            <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
              <g id="App" transform="translate(-282.000000, -478.000000)" fill="#837373" fill-rule="nonzero">
                <g id="Button.button.button--prev" transform="translate(300.000000, 511.500000) scale(-1, -1) translate(-300.000000, -511.500000) translate(282.000000, 478.000000)">
                  <path d="M2.7,66.561337 C2.07931034,67.146221 1.0862069,67.146221 0.465517241,66.561337 C-0.155172414,65.9456697 -0.155172414,64.9606019 0.465517241,64.3449345 L31.562069,33.5 L0.465517241,2.65506547 C-0.155172414,2.03939812 -0.155172414,1.05433035 0.465517241,0.438662991 C1.0862069,-0.146220997 2.07931034,-0.146220997 2.7,0.438662991 L36,33.5 L2.7,66.561337 Z"
                    id="Shape"></path>
                </g>
              </g>
            </g>
          </svg>
        </button>
      `;
    }
  });
}
