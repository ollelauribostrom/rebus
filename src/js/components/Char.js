import { createComponent } from '../mini';

export function Char(props) {
  return createComponent({
    props,
    render({ current, rebuses, wordIndex, charIndex }) {
      const rebus = rebuses[current];
      const previousWords = rebus.words.slice(0, wordIndex).join('');
      const index = wordIndex > 0 ? previousWords.length + charIndex : charIndex;
      const value = rebus.input[index] || '';
      return `
        <input
          type="text"
          maxlength="1"
          class="word__char"
          placeholder=" "
          value="${value}"
          ${rebus.isAnswered ? 'disabled' : ''}
        >`;
    }
  });
}
