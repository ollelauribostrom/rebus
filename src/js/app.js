import * as Sentry from '@sentry/browser';
import { render } from './mini';
import { App } from './components/app';
import { Logo } from './components/logo';
import { GithubCorner } from './components/github-corner';
import { ChangeButton } from './components/change-button';
import { Rebus } from './components/rebus';

import { actions } from './store';
import '../css/main.css';

export function registerListeners() {
  document.addEventListener('keyup', event => {
    const key = event.key || event.keyCode; // For older browser support
    if (key === 'ArrowRight' || key === 39) {
      actions.next();
    }
    if (key === 'ArrowLeft' || key === 37) {
      actions.prev();
    }
    if (key === 'Enter' || key === 13) {
      actions.shake();
    }
  });
}

export function setCurrentFromURL() {
  const params = new URLSearchParams(window.location.search);
  const id = Number(params.get('rebus'));
  actions.setCurrent(id);
}

export function init() {
  try {
    return render(
      App(
        Logo(),
        GithubCorner({ url: 'https://github.com/ollelauribostrom/rebus' }),
        ChangeButton({
          className: 'change-button--prev',
          onClick: () => actions.prev()
        }),
        Rebus({
          charInput: (input, wordIndex, charIndex) => {
            const confettiCanon = document.querySelector('.confetti-canon');
            actions.setInput(input, wordIndex, charIndex);
            actions.check(confettiCanon);
          }
        }),
        ChangeButton({
          className: 'change-button--next',
          onClick: () => actions.next()
        })
      ),
      document.querySelector('.root')
    );
  } catch (err) {
    return Sentry.captureException(err);
  }
}

if (!global || !global.isTestRun) {
  Sentry.init({ dsn: 'https://8f025bee12e84d9b8a16e9c3b9155ce8@sentry.io/1300214' });
  init();
  registerListeners();
  setCurrentFromURL();
}
