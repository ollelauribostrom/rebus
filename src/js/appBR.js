import { clickReset } from './components/ResetButton';

export function loadButtons(resetIdiom) {
  clickReset(resetIdiom);
  clickbr();
  clicken();
}

// BUTTON [PT-BR] - LOAD BUTTON
export function clickbr() {
  // CONDITION: if init as [pt-br] rebus, change design and hide button [ptbr]
  // ELSE: hide button [english]
  checkIsPTBR();

  const e = document.getElementById('button-ptbr'); // get button [pt-br]

  // HOVER FUNCTION: cursor (hover) [pt-br]
  mouseHoverPTBR(e);

  // CLICK FUNCTION: click on button [pt-br]
  e.addEventListener('click', () => {
    localStorage.setItem('flagBR', 'true');
  });
  // FLAG: now [pt-br] rebus
}

// BUTTON [EN] - LOAD BUTTON
export function clicken() {
  const a = document.getElementById('button-en'); // get button [english]

  // HOVER FUNCTION: cursor (hover) [english]
  mouseHoverEN(a);

  // CLICK FUNCTION: click on button [english]
  a.addEventListener('click', () => {
    localStorage.setItem('flagBR', 'false');
  });
  // FLAG: now [english / original] rebus
}

function mouseHoverPTBR(e) {
  e.addEventListener('mouseover', () => {
    document.getElementById('button-text-id-br').classList.add('button-text-show');
  });
  e.addEventListener('mouseout', () => {
    document.getElementById('button-text-id-br').classList.remove('button-text-show');
  });
}

function mouseHoverEN(a) {
  a.addEventListener('mouseover', () => {
    document.getElementById('button-text-id-en').classList.add('button-text-show');
  });
  a.addEventListener('mouseout', () => {
    document.getElementById('button-text-id-en').classList.remove('button-text-show');
  });
}

export function checkResetPTBR() {
  let resetIdiom = 'reset-english';
  if (localStorage.getItem('flagBR') === 'true') resetIdiom = 'reset-ptbr';
  return resetIdiom;
}

export function checkIsPTBR() {
  if (localStorage.getItem('flagBR') === 'true') {
    return (
      document.querySelector('.root').classList.add('rootbr'),
      document.getElementById('button-ptbr').classList.add('button-spam')
    );
  }
  return document.getElementById('button-en').classList.add('button-spam');
}
