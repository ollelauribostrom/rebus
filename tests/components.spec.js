import { render } from '../src/js/mini';
import { App, Word, Char, Logo, GithubCorner, ChangeButton, Rebus } from '../src/js/components';

jest.mock('../src/js/store', () => ({
  connect: arg => arg
}));

const mockState = {
  current: 0,
  animation: 'none',
  rebuses: [
    {
      symbols: ['😍', '👍', '😍'],
      words: ['one', 'two'],
      input: [...Array(6)],
      isAnswered: false
    }
  ]
};

describe('Tests for components', () => {
  describe('App', () => {
    it('renders correctly', () => {
      const wrapper = App(Logo());
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.render()).toMatchSnapshot();
    });
  });
  describe('Rebus', () => {
    it('renders correctly', () => {
      const props = mockState;
      const wrapper = Rebus(props);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.render(props)).toMatchSnapshot();
    });
    it('renders correctly (when rebus is answered)', () => {
      const props = { ...mockState };
      props.rebuses[0].isAnswered = true;
      const wrapper = Rebus(props);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.render(props)).toMatchSnapshot();
    });
    it('renders correctly (with animation class)', () => {
      const props = { ...mockState, animation: 'flip-vertical-right' };
      const wrapper = Rebus(props);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.render(props)).toMatchSnapshot();
    });
  });
  describe('Word', () => {
    it('renders correctly', () => {
      const props = { word: 'one', wordIndex: 0, charInput: jest.fn() };
      const wrapper = Word(props);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.render(props)).toMatchSnapshot();
    });
    it('correctly registers onInput handlers for each char', () => {
      const onInput = jest.fn();
      const props = { ...mockState, word: 'one', wordIndex: 0, charInput: onInput };
      const root = document.createElement('div');
      render(Word(props), root);
      const inputElement = root.querySelector('input');
      inputElement.value = 'T';
      inputElement.dispatchEvent(new Event('input'));
      expect(onInput).toHaveBeenCalledWith('T', 0, 0);
    });
    it('jumps to next simbling when a letter is pressed', () => {
      const onInput = jest.fn();
      const props = { ...mockState, word: 'one', wordIndex: 0, charInput: onInput };
      const root = document.createElement('div');
      render(Word(props), root);
      const inputs = root.querySelectorAll('input');
      inputs[0].focus();
      inputs[0].value = 'T';
      inputs[0].dispatchEvent(new Event('input'));
      expect(inputs[1] === document.activeElement).toEqual(true);
    });
    it('it remains on the same node when a character that is not a letter is pressed', () => {
      const onInput = jest.fn();
      const props = { ...mockState, word: 'one', wordIndex: 0, charInput: onInput };
      const root = document.createElement('div');
      render(Word(props), root);
      const inputs = root.querySelectorAll('input');
      inputs[0].focus();
      inputs[0].value = '!';
      inputs[0].dispatchEvent(new Event('input'));
      expect(inputs[0] === document.activeElement).toEqual(true);
    });
  });
  describe('Char', () => {
    it('renders correctly (without value)', () => {
      const props = { ...mockState, wordIndex: 0, charIndex: 1 };
      const wrapper = Char(props);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.render(props)).toMatchSnapshot();
    });
    it('renders correctly (with value and single word)', () => {
      const props = { ...mockState, wordIndex: 0, charIndex: 1 };
      props.rebuses[0].input[1] = 'c';
      const wrapper = Char(props);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.render(props)).toMatchSnapshot();
    });
    it('renders correctly (with value and multiple words)', () => {
      const props = { ...mockState, wordIndex: 1, charIndex: 1 };
      props.rebuses[0].input[4] = 'c';
      const wrapper = Char(props);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.render(props)).toMatchSnapshot();
    });
  });
  describe('ChangeButton', () => {
    it('renders correctly (without className prop)', () => {
      const props = {};
      const wrapper = ChangeButton(props);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.render(props)).toMatchSnapshot();
    });
    it('renders correctly (with className prop)', () => {
      const props = { className: 'test' };
      const wrapper = ChangeButton(props);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.render(props)).toMatchSnapshot();
    });
  });
  describe('GithubCorner', () => {
    it('renders correctly', () => {
      const props = { url: 'test' };
      const wrapper = GithubCorner(props);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.render(props)).toMatchSnapshot();
    });
  });
  describe('Logo', () => {
    it('renders correctly', () => {
      const wrapper = Logo();
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.render()).toMatchSnapshot();
    });
  });
});
