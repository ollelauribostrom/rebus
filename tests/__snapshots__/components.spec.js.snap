// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for components App renders correctly 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "componentDidMount": undefined,
      "componentDidUpdate": undefined,
      "props": Object {},
      "render": [Function],
      "update": [Function],
    },
  ],
  "componentDidMount": undefined,
  "componentDidUpdate": undefined,
  "props": Object {},
  "render": [Function],
  "update": [Function],
}
`;

exports[`Tests for components App renders correctly 2`] = `
"
        <div class=\\"app\\">
          <children>
        </div>
      "
`;

exports[`Tests for components ChangeButton renders correctly (with className prop) 1`] = `
Object {
  "children": Array [],
  "componentDidMount": undefined,
  "componentDidUpdate": undefined,
  "props": Object {
    "className": "test",
  },
  "render": [Function],
  "update": [Function],
}
`;

exports[`Tests for components ChangeButton renders correctly (with className prop) 2`] = `
"
        <button class=\\"change-button test\\">
          <svg width=\\"36px\\" height=\\"67px\\" viewBox=\\"0 0 36 67\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\"
            xmlns:xlink=\\"http://www.w3.org/1999/xlink\\">
            <defs></defs>
            <g id=\\"Page-1\\" stroke=\\"none\\" stroke-width=\\"1\\" fill=\\"none\\" fill-rule=\\"evenodd\\">
              <g id=\\"App\\" transform=\\"translate(-282.000000, -478.000000)\\" fill=\\"#837373\\" fill-rule=\\"nonzero\\">
                <g id=\\"Button.button.button--prev\\" transform=\\"translate(300.000000, 511.500000) scale(-1, -1) translate(-300.000000, -511.500000) translate(282.000000, 478.000000)\\">
                  <path d=\\"M2.7,66.561337 C2.07931034,67.146221 1.0862069,67.146221 0.465517241,66.561337 C-0.155172414,65.9456697 -0.155172414,64.9606019 0.465517241,64.3449345 L31.562069,33.5 L0.465517241,2.65506547 C-0.155172414,2.03939812 -0.155172414,1.05433035 0.465517241,0.438662991 C1.0862069,-0.146220997 2.07931034,-0.146220997 2.7,0.438662991 L36,33.5 L2.7,66.561337 Z\\"
                    id=\\"Shape\\"></path>
                </g>
              </g>
            </g>
          </svg>
        </button>
      "
`;

exports[`Tests for components ChangeButton renders correctly (without className prop) 1`] = `
Object {
  "children": Array [],
  "componentDidMount": undefined,
  "componentDidUpdate": undefined,
  "props": Object {},
  "render": [Function],
  "update": [Function],
}
`;

exports[`Tests for components ChangeButton renders correctly (without className prop) 2`] = `
"
        <button class=\\"change-button \\">
          <svg width=\\"36px\\" height=\\"67px\\" viewBox=\\"0 0 36 67\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\"
            xmlns:xlink=\\"http://www.w3.org/1999/xlink\\">
            <defs></defs>
            <g id=\\"Page-1\\" stroke=\\"none\\" stroke-width=\\"1\\" fill=\\"none\\" fill-rule=\\"evenodd\\">
              <g id=\\"App\\" transform=\\"translate(-282.000000, -478.000000)\\" fill=\\"#837373\\" fill-rule=\\"nonzero\\">
                <g id=\\"Button.button.button--prev\\" transform=\\"translate(300.000000, 511.500000) scale(-1, -1) translate(-300.000000, -511.500000) translate(282.000000, 478.000000)\\">
                  <path d=\\"M2.7,66.561337 C2.07931034,67.146221 1.0862069,67.146221 0.465517241,66.561337 C-0.155172414,65.9456697 -0.155172414,64.9606019 0.465517241,64.3449345 L31.562069,33.5 L0.465517241,2.65506547 C-0.155172414,2.03939812 -0.155172414,1.05433035 0.465517241,0.438662991 C1.0862069,-0.146220997 2.07931034,-0.146220997 2.7,0.438662991 L36,33.5 L2.7,66.561337 Z\\"
                    id=\\"Shape\\"></path>
                </g>
              </g>
            </g>
          </svg>
        </button>
      "
`;

exports[`Tests for components Char renders correctly (with value and multiple words) 1`] = `
Object {
  "children": Array [],
  "componentDidMount": undefined,
  "componentDidUpdate": undefined,
  "props": Object {
    "animation": "none",
    "charIndex": 1,
    "current": 0,
    "incorrectAnswerCount": 0,
    "rebuses": Array [
      Object {
        "hint": "hint",
        "input": Array [
          undefined,
          undefined,
          undefined,
          undefined,
          "c",
          undefined,
        ],
        "isAnswered": false,
        "symbols": Array [
          "üòç",
          "üëç",
          "üòç",
        ],
        "words": Array [
          "one",
          "two",
        ],
      },
    ],
    "wordIndex": 1,
  },
  "render": [Function],
  "update": [Function],
}
`;

exports[`Tests for components Char renders correctly (with value and multiple words) 2`] = `
"
        <input
          type=\\"text\\"
          maxlength=\\"1\\"
          class=\\"word__char\\"
          placeholder=\\" \\"
          value=\\"c\\"
          
        >"
`;

exports[`Tests for components Char renders correctly (with value and single word) 1`] = `
Object {
  "children": Array [],
  "componentDidMount": undefined,
  "componentDidUpdate": undefined,
  "props": Object {
    "animation": "none",
    "charIndex": 1,
    "current": 0,
    "incorrectAnswerCount": 0,
    "rebuses": Array [
      Object {
        "hint": "hint",
        "input": Array [
          undefined,
          "c",
          undefined,
          undefined,
          undefined,
          undefined,
        ],
        "isAnswered": false,
        "symbols": Array [
          "üòç",
          "üëç",
          "üòç",
        ],
        "words": Array [
          "one",
          "two",
        ],
      },
    ],
    "wordIndex": 0,
  },
  "render": [Function],
  "update": [Function],
}
`;

exports[`Tests for components Char renders correctly (with value and single word) 2`] = `
"
        <input
          type=\\"text\\"
          maxlength=\\"1\\"
          class=\\"word__char\\"
          placeholder=\\" \\"
          value=\\"c\\"
          
        >"
`;

exports[`Tests for components Char renders correctly (without value) 1`] = `
Object {
  "children": Array [],
  "componentDidMount": undefined,
  "componentDidUpdate": undefined,
  "props": Object {
    "animation": "none",
    "charIndex": 1,
    "current": 0,
    "incorrectAnswerCount": 0,
    "rebuses": Array [
      Object {
        "hint": "hint",
        "input": Array [
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ],
        "isAnswered": false,
        "symbols": Array [
          "üòç",
          "üëç",
          "üòç",
        ],
        "words": Array [
          "one",
          "two",
        ],
      },
    ],
    "wordIndex": 0,
  },
  "render": [Function],
  "update": [Function],
}
`;

exports[`Tests for components Char renders correctly (without value) 2`] = `
"
        <input
          type=\\"text\\"
          maxlength=\\"1\\"
          class=\\"word__char\\"
          placeholder=\\" \\"
          value=\\"\\"
          
        >"
`;

exports[`Tests for components GithubCorner renders correctly 1`] = `
Object {
  "children": Array [],
  "componentDidMount": undefined,
  "componentDidUpdate": undefined,
  "props": Object {
    "url": "test",
  },
  "render": [Function],
  "update": [Function],
}
`;

exports[`Tests for components GithubCorner renders correctly 2`] = `
"
        <a href=\\"test\\" class=\\"github-corner\\" aria-label=\\"View source on Github\\">
          <svg width=\\"80\\" height=\\"80\\" viewBox=\\"0 0 250 250\\" aria-hidden=\\"true\\">
            <path d=\\"M0,0 L115,115 L130,115 L142,142 L250,250 L250,0 Z\\"></path>
            <path d=\\"M128.3,109.0 C113.8,99.7 119.0,89.6 119.0,89.6 C122.0,82.7 120.5,78.6 120.5,78.6 C119.2,72.0 123.4,76.3 123.4,76.3 C127.3,80.9 125.5,87.3 125.5,87.3 C122.9,97.6 130.6,101.9 134.4,103.2\\"
              fill=\\"currentColor\\" style=\\"transform-origin: 130px 106px;\\" class=\\"octo-arm\\"></path>
            <path d=\\"M115.0,115.0 C114.9,115.1 118.7,116.5 119.8,115.4 L133.7,101.6 C136.9,99.2 139.9,98.4 142.2,98.6 C133.8,88.0 127.5,74.4 143.8,58.0 C148.5,53.4 154.0,51.2 159.7,51.0 C160.3,49.4 163.2,43.6 171.4,40.1 C171.4,40.1 176.1,42.5 178.8,56.2 C183.1,58.6 187.2,61.8 190.9,65.4 C194.5,69.0 197.7,73.2 200.1,77.6 C213.8,80.2 216.3,84.9 216.3,84.9 C212.7,93.1 206.9,96.0 205.4,96.6 C205.1,102.4 203.0,107.8 198.3,112.5 C181.9,128.9 168.3,122.5 157.7,114.1 C157.9,116.9 156.7,120.9 152.7,124.9 L141.0,136.5 C139.8,137.7 141.6,141.9 141.8,141.8 Z\\"
              fill=\\"currentColor\\" class=\\"octo-body\\"></path>
          </svg>
        </a>
      "
`;

exports[`Tests for components Hint renders correctly (when incorrect answer count is less than max incorrect answer count) 1`] = `
Object {
  "children": Array [],
  "componentDidMount": undefined,
  "componentDidUpdate": undefined,
  "props": Object {
    "animation": "none",
    "current": 0,
    "incorrectAnswerCount": 1,
    "rebuses": Array [
      Object {
        "hint": "hint",
        "input": Array [
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ],
        "isAnswered": false,
        "symbols": Array [
          "üòç",
          "üëç",
          "üòç",
        ],
        "words": Array [
          "one",
          "two",
        ],
      },
    ],
  },
  "render": [Function],
  "update": [Function],
}
`;

exports[`Tests for components Hint renders correctly (when incorrect answer count is less than max incorrect answer count) 2`] = `
" 
        <span></span>
        "
`;

exports[`Tests for components Hint renders correctly (when incorrect answer count is more than max incorrect answer count) 1`] = `
Object {
  "children": Array [],
  "componentDidMount": undefined,
  "componentDidUpdate": undefined,
  "props": Object {
    "animation": "none",
    "current": 0,
    "incorrectAnswerCount": 4,
    "rebuses": Array [
      Object {
        "hint": "hint",
        "input": Array [
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ],
        "isAnswered": false,
        "symbols": Array [
          "üòç",
          "üëç",
          "üòç",
        ],
        "words": Array [
          "one",
          "two",
        ],
      },
    ],
  },
  "render": [Function],
  "update": [Function],
}
`;

exports[`Tests for components Hint renders correctly (when incorrect answer count is more than max incorrect answer count) 2`] = `
" 
        <span class=\\"rebus__hint\\">
                üí° hint
              </span>
        "
`;

exports[`Tests for components Logo renders correctly 1`] = `
Object {
  "children": Array [],
  "componentDidMount": undefined,
  "componentDidUpdate": undefined,
  "props": Object {},
  "render": [Function],
  "update": [Function],
}
`;

exports[`Tests for components Logo renders correctly 2`] = `
"
      <svg class=\\"logo\\" width=\\"52px\\" height=\\"52px\\" viewBox=\\"0 0 52 52\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\"
        xmlns:xlink=\\"http://www.w3.org/1999/xlink\\">
        <g id=\\"Page-1\\" stroke=\\"none\\" stroke-width=\\"1\\" fill=\\"none\\" fill-rule=\\"evenodd\\">
          <g id=\\"App\\" transform=\\"translate(-49.000000, -41.000000)\\" fill=\\"#fff\\">
            <g id=\\".header\\" transform=\\"translate(43.000000, 32.000000)\\">
              <path d=\\"M25.92,48.64 L22.24,48.64 L22.24,20.4 L30.96,20.4 C33.093344,20.4 34.7066612,20.7066636 35.8,21.32 C36.8933388,21.9333364 37.7466636,22.8266608 38.36,24 C38.9733364,25.1733392 39.28,26.4533264 39.28,27.84 C39.28,28.9600056 39.0666688,30.0133284 38.64,31 C38.2133312,31.9866716 37.6133372,32.8933292 36.84,33.72 C36.0666628,34.5466708 34.9066744,35.2799968 33.36,35.92 C33.9466696,36.2400016 34.3999984,36.5599984 34.72,36.88 C35.0400016,37.2000016 35.4133312,37.7066632 35.84,38.4 C36.2666688,39.0933368 36.6399984,39.706664 36.96,40.24 L41.44,48.64 L37.28,48.64 L33.36,41.2 C32.3466616,39.3866576 31.0933408,37.8400064 29.6,36.56 L25.92,36.56 L25.92,48.64 Z M25.92,33.68 L29.52,33.68 C31.0666744,33.68 32.2266628,33.4400024 33,32.96 C33.7733372,32.4799976 34.3866644,31.8000044 34.84,30.92 C35.2933356,30.0399956 35.52,29.0933384 35.52,28.08 C35.52,27.1733288 35.3333352,26.3733368 34.96,25.68 C34.5866648,24.9866632 34.0533368,24.4000024 33.36,23.92 C32.6666632,23.4399976 31.4666752,23.2 29.76,23.2 L25.92,23.2 L25.92,33.68 Z M57.84,60.88 L52.8,60.96 L52.8,58.88 L55.92,58.88 L55.92,55.6 L57.92,55.6 L57.84,60.88 Z M49.76,60.88 L43.36,60.88 L43.36,58.88 L49.76,58.88 L49.76,60.88 Z M39.84,60.88 L33.6,60.88 L33.6,58.88 L39.84,58.88 L39.84,60.88 Z M29.92,60.88 L23.68,60.88 L23.68,58.88 L29.92,58.88 L29.92,60.88 Z M20.64,60.88 L14.32,60.88 L14.4,58.88 L20.64,58.96 L20.64,60.88 Z M11.36,60.88 L6,60.88 L6.16,55.52 L8.16,55.52 L8.16,58.88 L11.36,58.88 L11.36,60.88 Z M8.16,52.24 L6.08,52.24 L6.08,46.08 L8.16,46.16 L8.16,52.24 Z M8.16,42.88 L6,42.88 L6.08,36.8 L8.16,36.8 L8.16,42.88 Z M8.16,33.68 L6,33.68 L6.08,27.44 L8.16,27.36 L8.16,33.68 Z M8.16,24.4 L6,24.4 L6.16,17.92 L8.16,17.92 L8.16,24.4 Z M8.16,14.96 L6.16,14.96 L6.16,10 L11.36,9.84 L11.36,11.92 L8.24,11.92 L8.16,14.96 Z M14.4,11.92 L14.4,9.92 L20.72,10 L20.64,11.92 L14.4,11.92 Z M23.76,11.92 L23.68,10 L30,9.84 L30.08,12 L23.76,11.92 Z M33.68,12 L33.68,10 L40,10 L40,11.92 L33.68,12 Z M43.36,11.92 L43.36,10 L50,10 L49.92,12 L43.36,11.92 Z M52.96,10 L58,10 L57.92,15.12 L55.92,15.12 L55.92,11.92 L52.88,12.08 L52.96,10 Z M57.92,18.08 L57.92,24.48 L55.92,24.48 L55.92,18 L57.92,18.08 Z M57.92,27.44 L57.92,33.76 L56,33.68 L55.92,27.44 L57.92,27.44 Z M57.84,36.96 L57.84,42.96 L55.92,42.96 L55.92,36.88 L57.84,36.96 Z M57.76,46.16 L57.76,52.32 L55.92,52.4 L55.92,46.16 L57.76,46.16 Z\\"
                id=\\"Logo.logo\\"></path>
            </g>
          </g>
        </g>
      </svg>
      "
`;

exports[`Tests for components ProgressBar renders correctly (without rebuses) 1`] = `
Object {
  "children": Array [],
  "componentDidMount": undefined,
  "componentDidUpdate": undefined,
  "props": Object {
    "rebuses": Array [],
  },
  "render": [Function],
  "update": [Function],
}
`;

exports[`Tests for components ProgressBar renders correctly (without rebuses) 2`] = `
" 
        <progress class=\\"progress-bar\\" value=\\"0\\" max=\\"0\\">
        </progress> 
        "
`;

exports[`Tests for components ProgressBar renders correctly 1`] = `
Object {
  "children": Array [],
  "componentDidMount": undefined,
  "componentDidUpdate": undefined,
  "props": Object {
    "animation": "none",
    "current": 0,
    "incorrectAnswerCount": 0,
    "rebuses": Array [
      Object {
        "hint": "hint",
        "input": Array [
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ],
        "isAnswered": false,
        "symbols": Array [
          "üòç",
          "üëç",
          "üòç",
        ],
        "words": Array [
          "one",
          "two",
        ],
      },
    ],
  },
  "render": [Function],
  "update": [Function],
}
`;

exports[`Tests for components ProgressBar renders correctly 2`] = `
" 
        <progress class=\\"progress-bar\\" value=\\"0\\" max=\\"1\\">
        </progress> 
        "
`;

exports[`Tests for components Rebus renders correctly (when rebus is answered) 1`] = `
Object {
  "children": Array [],
  "componentDidMount": [Function],
  "componentDidUpdate": [Function],
  "props": Object {
    "animation": "none",
    "current": 0,
    "incorrectAnswerCount": 0,
    "rebuses": Array [
      Object {
        "hint": "hint",
        "input": Array [
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ],
        "isAnswered": true,
        "symbols": Array [
          "üòç",
          "üëç",
          "üòç",
        ],
        "words": Array [
          "one",
          "two",
        ],
      },
    ],
  },
  "render": [Function],
  "update": [Function],
}
`;

exports[`Tests for components Rebus renders correctly (when rebus is answered) 2`] = `
"
          <div class=\\"rebus rebus--answered animation--none\\">
            <div class=\\"rebus__header\\">
              <span>1/1</span>
            </div>
            <span class=\\"rebus__symbols\\">üòç üëç üòç</span>
            <div class=\\"rebus__words\\">
              <children>
            </div>
          </div>
        "
`;

exports[`Tests for components Rebus renders correctly (with animation class) 1`] = `
Object {
  "children": Array [],
  "componentDidMount": [Function],
  "componentDidUpdate": [Function],
  "props": Object {
    "animation": "flip-vertical-right",
    "current": 0,
    "incorrectAnswerCount": 0,
    "rebuses": Array [
      Object {
        "hint": "hint",
        "input": Array [
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ],
        "isAnswered": false,
        "symbols": Array [
          "üòç",
          "üëç",
          "üòç",
        ],
        "words": Array [
          "one",
          "two",
        ],
      },
    ],
  },
  "render": [Function],
  "update": [Function],
}
`;

exports[`Tests for components Rebus renders correctly (with animation class) 2`] = `
"
          <div class=\\"rebus  animation--flip-vertical-right\\">
            <div class=\\"rebus__header\\">
              <span>1/1</span>
            </div>
            <span class=\\"rebus__symbols\\">üòç üëç üòç</span>
            <div class=\\"rebus__words\\">
              <children>
            </div>
          </div>
        "
`;

exports[`Tests for components Rebus renders correctly 1`] = `
Object {
  "children": Array [],
  "componentDidMount": [Function],
  "componentDidUpdate": [Function],
  "props": Object {
    "animation": "none",
    "current": 0,
    "incorrectAnswerCount": 0,
    "rebuses": Array [
      Object {
        "hint": "hint",
        "input": Array [
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ],
        "isAnswered": false,
        "symbols": Array [
          "üòç",
          "üëç",
          "üòç",
        ],
        "words": Array [
          "one",
          "two",
        ],
      },
    ],
  },
  "render": [Function],
  "update": [Function],
}
`;

exports[`Tests for components Rebus renders correctly 2`] = `
"
          <div class=\\"rebus  animation--none\\">
            <div class=\\"rebus__header\\">
              <span>1/1</span>
            </div>
            <span class=\\"rebus__symbols\\">üòç üëç üòç</span>
            <div class=\\"rebus__words\\">
              <children>
            </div>
          </div>
        "
`;

exports[`Tests for components Word renders correctly 1`] = `
Object {
  "children": Array [],
  "componentDidMount": undefined,
  "componentDidUpdate": undefined,
  "props": Object {
    "charInput": [MockFunction],
    "word": "one",
    "wordIndex": 0,
  },
  "render": [Function],
  "update": [Function],
}
`;

exports[`Tests for components Word renders correctly 2`] = `
"
          <div class=\\"word\\">
            <children>
          </div>
        "
`;
